File: Backend/src/AudioAtlasDomain/Genre.cs
namespace Domain.Genre;

public class Genre
{
    public required string Id { get; set; }
    public required string Name { get; set; }
    public string Description { get; set; }
    public required ICollection<string> Origin { get; set; } = new List<string>();
    public ICollection<string>? Aliases { get; set; }
    public ICollection<Genre> RelatedGenres { get; set; } = new List<Genre>();
}

File: frontend/app/types/genre.ts
import type { GenreAlias } from "./genreAlias";
import type { Country, ContributorSummary } from "./country"

export interface Genre {
  id: string
  name: string
  description?: string
  startYear?: number
  isSensitive:boolean
  playlistLink?: string
  sensitiveDescription?: string
  countries: Country[]
  aliases?: GenreAlias[]
  similarGenres: Genre[]
  subGenres: Genre[]
  parentGenres: Genre[]
  contributors: ContributorSummary[]
}
